### PR TITLE
Add tile object support

### PR DESCRIPTION
**Pull request by vinnydiehl**
Since the scope of `ObjectLayer#render_debug` is no longer only for debugging, I have renamed it to `ObjectLayer#render`. I also tweaked its parameters so it's nicer to call and defaults to `args.outputs.primitives`.

The header of `Sprite::from_tiled` now takes optional width and height params:
```rb
def self.from_tiled(x, y, tile, w: nil, h: nil)
```
This way avoids breaking existing implementations, but it's a bit ugly. Assuming we're still in the "move fast and break shit" stage of development, we might consider putting the `tile` parameter first, so that x/y/w/h are right next to each other. Your call, this doesn't bother me *that* much.

### CODE REVIEW

File: lib/tiled/object_layer.rb [hunk 1]
```diff
@@ -33,11 +33,13 @@ def sprites
...
     # Renders the objects to the `outputs_layer`.
-    # @param args `args` from `tick` method
-    # @param output_layer one of `args.outputs`, works with `primitives` and `debug`
-    def render_debug(args, outputs_layer)
+    # @param args [GTK::Args] `args` from `tick` method
+    # @param target [Symbol] the output target, either :primitives or :debug
+    def render(args, target=:primitives)
       return unless visible?
 
+      outputs_layer = args.outputs.send(target)
```
Comment (wildfiler) [inline lib/tiled/object_layer.rb]: This will not work if I pass a render target here.

File: lib/tiled/sprite.rb [hunk 2]
```diff
@@ -18,13 +18,13 @@ def primitive_marker
     end
 
     # @return [Tiled::Sprite] return sprite object from `Tiled::Tile` object.
-    def self.from_tiled(x, y, tile)
+    def self.from_tiled(x, y, tile, w: nil, h: nil)
```
Comment (wildfiler) [inline lib/tiled/sprite.rb]: I agree, it makes sense to move the `tile` argument to the first place. 

Or it would be even better to replace all arguments with keyword arguments.

File: lib/tiled/tiled_object.rb [hunk 3]
```diff
@@ -4,13 +4,15 @@ class TiledObject
...
-    attributes :id, :x, :y, :width, :height, :shape, :points
+    attributes :id, :gid, :x, :y, :width, :height, :type, :points
 
     def initialize(map, attrs, children)
       @map = map
-      attributes.add(id: attrs['id'], shape: children.any? ? children.first[:name].to_sym : :rectangle)
+      attributes.add(id: attrs['id'], gid: attrs['gid']&.to_i)
 
-      if shape == :polygon
+      attributes.add(type: children.any? ? children.first[:name].to_sym : gid ? :tile : :rectangle)
```
Comment (wildfiler) [inline lib/tiled/tiled_object.rb]: `type` is an object attribute in TMX format, so we should not replace it with our value. Maybe `object_type`?

Very hard to parse this expression, I would prefer something like this:
```ruby
def initialize(map, attrs, children)
  # ...
  attributes.add(object_type: detect_type(attrs))
  # ...
end

def detect_type(attrs, children)
  if children.any?
    children.first[:name].to_sym
  elsif attrs['gid']
    :tile
  else
    :rectangle
  end
end
```

**Comment by vinnydiehl** [comment]
Done, done, and done.

**Review by wildfiler — APPROVED** [review]
